versioned/tiered/tiered-impl/src/main/java/org/projectnessie/versioned/store/TracingStore.java

```diff
@@ -0,0 +1,248 @@
+        .withTag("nessie.store.operation", "Delete")
+        .withTag("nessie.store.value-type", safeName(type))
+        .withTag("nessie.store.id", safeToString(id))
+        .startActive(true)) {
+      try {
+        return store.delete(type, id, condition);
+      } catch (RuntimeException e) {
+        throw traceRuntimeException(scope, e);
+      }
+    }
+  }
+
+  @Override
+  public void save(List<SaveOp<?>> ops) {
+    try (Scope scope = createSpan("Store.save")
+        .withTag("nessie.store.operation", "Save")
+        .withTag("nessie.store.num-ops", safeSize(ops))
+        .startActive(true)) {
+      try {
+        scope.span().log(ops.stream().collect(Collectors.groupingBy(
+            op -> String.format("nessie.store.save.%s.ids", op.getType().name()),
+            Collectors.mapping(op -> op.getId().toString(), Collectors.joining(", "))
+        )));
+
+        store.save(ops);
+      } catch (RuntimeException e) {
+        throw traceRuntimeException(scope, e);
+      }
+    }
+  }
+
+  @Override
+  public <C extends BaseValue<C>> void loadSingle(
+      ValueType<C> type, Id id, C consumer) {
+    try (Scope scope = createSpan("Store.loadSingle")
+        .withTag("nessie.store.operation", "LoadSingle")
+        .withTag("nessie.store.value-type", safeName(type))
+        .withTag("nessie.store.id", safeToString(id))
+        .startActive(true)) {
+      try {
+        store.loadSingle(type, id, consumer);
+      } catch (RuntimeException e) {
+        throw traceRuntimeException(scope, e);
+      }
+    }
+  }
+
+  @Override
+  public <C extends BaseValue<C>> boolean update(
+      ValueType<C> type, Id id, UpdateExpression update,
+      Optional<ConditionExpression> condition,
+      Optional<BaseValue<C>> consumer) throws NotFoundException {
+    try (Scope scope = createSpan("Store.update")
+        .withTag("nessie.store.operation", "Update")
+        .withTag("nessie.store.value-type", safeName(type))
+        .withTag("nessie.store.id", safeToString(id))
+        .withTag("nessie.store.update", safeToString(update))
+        .withTag("nessie.store.condition", safeToString(condition))
+        .startActive(true)) {
+      try {
+        return store.update(type, id, update, condition, consumer);
+      } catch (RuntimeException e) {
+        throw traceRuntimeException(scope, e);
+      }
+    }
+  }
+
+  @Override
+  public <C extends BaseValue<C>> Stream<Acceptor<C>> getValues(
+      ValueType<C> type) {
+    Scope scope = createSpan("Store.getValues")
+        .withTag("nessie.store.operation", "GetValues")
+        .withTag("nessie.store.value-type", type.name())
+        .startActive(true);
+    try {
+      return store.getValues(type).onClose(scope::close);
+    } catch (RuntimeException e) {
+      e = traceRuntimeException(scope, e);
+      scope.close();
+      throw e;
+    }
+  }
+
+  private static <C extends BaseValue<C>> String safeName(ValueType<C> type) {
```

Comment:
some dup code here and in version store. Possible to create a util?

Reply:
Yup, done. Also tried to "unify" the test-code stuff to reuse common things and to make the test-code a more readable.